### PR TITLE
Refresh Manifest Cache failed: Extension is not currently installed

### PR DESCRIPTION
**Pull request by brianteeman**
PR for #17604

Change the message to include the name of the extension.

I have no idea how to test this - sorry - only code review - unless someone knows how?

### CODE REVIEW

File: libraries/src/Installer/Installer.php [hunk 1]
```diff
@@ -833,7 +832,7 @@ public function refreshManifestCache($eid)
 			}
 		}
 
-		$this->abort(\JText::_('JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID'));
+		$this->abort(\JText::sprintf('JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID', $this->extension->name));
```
Comment (joomdonation) [inline libraries/src/Installer/Installer.php]: This is not correct as we don't have $this->extension->name available for using here

File: language/en-GB/en-GB.lib_joomla.ini [hunk 2]
```diff
@@ -531,8 +531,8 @@ JLIB_INSTALLER_ABORT_PLG_INSTALL_NO_FILE="Plugin %s: No plugin file specified."
 JLIB_INSTALLER_ABORT_PLG_INSTALL_ROLLBACK="Plugin %1$s: %2$s"
 JLIB_INSTALLER_ABORT_PLG_INSTALL_SQL_ERROR="Plugin %1$s: SQL error file %2$s"
 JLIB_INSTALLER_ABORT_PLG_UNINSTALL_SQL_ERROR="Plugin Uninstall: SQL error file %s"
-JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE="Refresh Manifest Cache failed: Extension is not currently installed."
-JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID="Refresh Manifest Cache failed: Extension is not valid."
+JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE="Refresh Manifest Cache failed: %s Extension is not currently installed."
+JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID="Refresh Manifest Cache failed: %s Extension is not valid."
```
Comment (joomdonation) [inline language/en-GB/en-GB.lib_joomla.ini]: This change need to be reverted (see my comment related to PHP code)

**Comment by joomdonation** [comment]
To test the first case, I had to alter data from database in #__extensions table for an extension, and then go to Extensions -> Manage, select that extension and press Refresh cache

I don't know how an extension can have state = -1, not installed (see this error message **Refresh Manifest Cache failed: com_banners Extension is not currently installed**)  (even in our state filter doesn't have this state). So maybe the code and language item for that case could be removed. But I am unsure.

**Comment by brianteeman** [comment]
> To test the first case, I had to alter data from database in #__extensions table for an extension, and then go to Extensions -> Manage, select that extension and press Refresh cache

When you say "alter data" what exactly do you mean? If I set the state to -1 then it doesnt appear in Extensions->Manage so i cant "refresh the cache"

> I don't know how an extension can have state = -1, 

Perhaps a discovered extension that hasnt been "installed" yet. Anyway it happens because the user in #17604 reported it

**Comment by joomdonation** [comment]
> If I set the state to -1 then it doesnt appear in Extensions->Manage so i cant "refresh the cache"

Go to Extensions -> Manage -> Manage to see the list of extensions first and select the one you want, then change state to -1 for that extension record in #__extensions table via phpmyadmin and then press Refresh Cache button

**Comment by brianteeman** [comment]
Thanks for the tip. Will to check it in the morning

**Comment by brianteeman** [comment]
@joomdonation Sorry I dont understand what the problem is with this PR. After using your test instructions (thanks) the error message is now correctly reported as 
> Refresh Manifest Cache failed: com_banners Extension is not currently installed.

Instead of 

> Refresh Manifest Cache failed: Extension is not currently installed.

**Comment by brianteeman** [comment]
conflicts resolved

**Comment by joomdonation** [comment]
@brianteeman The change you made in line 385 is not correct (It is the case empty $eid is passed to method call, so $this->extension->name will return an empty value and should not be used not be used in the language item)

```php 
$this->abort(\JText::sprintf('JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID', $this->extension->name));
```
In short, just revert the change you made to line 385 and the language item JLIB_INSTALLER_ABORT_REFRESH_MANIFEST_CACHE_VALID and you PR will be fine

If you want to test this case, just add this line of code at the beginning of the method:

$eid = 0;

Then check the message display after clicking on Refresh Cache button.

**Comment by brianteeman** [comment]
Sorry I completely misunderstood you. My fault for reading jut the email and not looking at the entire thing. Fix coming - thanks for your patience

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 545a70bc643027644ff695826d790a216349766d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19560">issues.joomla.org/tracker/joomla-cms/19560</a>.</sub>

**Comment by joomdonation** [comment]
Thanks, it's fine now. Better, please revert the comment line in line 809 which you removed, too

// Fetch the adapter

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9e07f8019e8023b4e63cc5b9dced1e471edf5e3d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19560">issues.joomla.org/tracker/joomla-cms/19560</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19560">issues.joomla.org/tracker/joomla-cms/19560</a>.</sub>

**Pull request merged**